src/TypeSpec.Extension/Emitter.Csharp/src/emitter.ts

```diff
@@ -1,205 +1,135 @@
+    if (options.skipSDKGeneration !== true) {
+        const configurationFilePath = resolvePath(
+            outputFolder,
+            configurationFileName
+        );
+        const configurations = JSON.parse(
+            fs.readFileSync(configurationFilePath, "utf-8")
+        );
+        //resolve shared folders based on generator path override
+        const resolvedSharedFolders: string[] = [];
+        const sharedFolders = [
+            resolvePath(options.csharpGeneratorPath, "..", "Generator.Shared"),
+            resolvePath(options.csharpGeneratorPath, "..", "Azure.Core.Shared")
+        ];
+        for (const sharedFolder of sharedFolders) {
+            resolvedSharedFolders.push(
+                path
+                    .relative(generatedFolder, sharedFolder)
+                    .replaceAll("\\", "/")
             );
-
-            if (options.skipSDKGeneration !== true) {
-                const csProjFile = resolvePath(
-                    outputFolder.endsWith("src")
-                        ? outputFolder
-                        : resolvePath(outputFolder, "src"),
-                    `${configurations["library-name"]}.csproj`
-                );
-                Logger.getInstance().info(`Checking if ${csProjFile} exists`);
-                const newProjectOption =
-                    options["new-project"] || !existsSync(csProjFile)
-                        ? "--new-project"
-                        : "";
-                const existingProjectOption = options["existing-project-folder"]
-                    ? `--existing-project-folder ${options["existing-project-folder"]}`
-                    : "";
-                const debugFlag = options.debug ?? false ? " --debug" : "";
-
-                const command = `dotnet --roll-forward Major ${resolvePath(
-                    options.csharpGeneratorPath
-                )} --project-path ${outputFolder} ${newProjectOption} ${existingProjectOption} --clear-output-folder ${
-                    options["clear-output-folder"]
-                }${debugFlag}`;
-                Logger.getInstance().info(command);
-
-                try {
-                    execSync(command, { stdio: "inherit" });
-                } catch (error: any) {
-                    if (error.message) Logger.getInstance().info(error.message);
-                    if (error.stderr) Logger.getInstance().error(error.stderr);
-                    if (error.stdout)
-                        Logger.getInstance().verbose(error.stdout);
-                    throw error;
-                }
-            }
-
-            if (!options["save-inputs"]) {
-                // delete
-                deleteFile(resolvePath(generatedFolder, tspOutputFileName));
-                deleteFile(resolvePath(generatedFolder, configurationFileName));
-            }
+        }
+        configurations["shared-source-folders"] = resolvedSharedFolders ?? [];
+        configurations["flavor"] =
+            options["flavor"] ??
```

Comment:
Why we need the logic of handling flavor here?  Is this not handled in common emitter?

Reply:
This is azure-specific logic(set the flavor according to the sdk package name) which is not included in common emitter.